File: matlab_scripts/kf_calc_Fx.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% F = kf_calcDFx(x) Calculates the Jacobian of the system dynamics equation f(x,u,t) 
%   
%   Author: C.C. de Visser, Delft University of Technology, 2013
%   email: c.c.devisser@tudelft.nl
%   Version: 1.0
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
function DFx = kf_calcDFx(t, x, u)
    
    n = size(x, 1);
    
    % Calculate Jacobian matrix of system dynamics
    DFx = zeros(4);
end
    

File: kf_calc_Fx.py
########################################################################
## Calculates the Jacobian of the system dynamics equation 
########################################################################
import numpy as np


def kf_calc_Fx(t, x, u):
    """
    Calculates the Jacobian of the system dynamics equation,
    n is number of states.
    
    Parameters
    ----------
    t : float
        
    x : numpy.ndarray (n,1)
        state vector

    u : numpy.ndarray
        input vector

    Returns
    -------
    DFx : numpy.ndarray (n,n)
        Jacobian of the system dynamics equation
        
    """
    n = x.size
    
    # calculate Jacobian matrix of system dynamics
    DFx = np.zeros([4, 4])
    
    return DFx
        